src/ctia/schemas/graphql/helpers.clj

```diff
@@ -104,16 +104,66 @@
   [msg value]
   (log/debug msg (pr-str value)))
 
+(defn get-selections [s]
+  (some-> s .getSelectionSet .getSelections))
+
+(defn get-selections-get [s]
+  (some-> s .getSelectionSet .get))
+
+(defn fragment-spread? [x]
+  (instance? graphql.language.FragmentSpread x))
+
+(defn resolve-fields [env all-fragments]
+  (let [selections (get-selections env)
+        fragments (filter fragment-spread? selections)
+        non-fragments (remove fragment-spread? selections)
+        non-fragment-fields (map #(.getName %) non-fragments)
+        fragment-names (->> fragments
+                            (map #(.getName %))
+                            (map keyword))
+        fragments (map #(get all-fragments %) fragment-names)
+        fragment-fields (mapcat #(get-selections %) fragments)
+        fragments-fields (map #(.getName %) fragment-fields)]
+    (map #(if (string? %)
+            (keyword %)
+            (keyword (.getName %)))
+         (concat non-fragment-fields
+                 fragment-fields))))
+
+(defn traverse [edges-node n]
+  (let [selections (get-selections edges-node)]
+    (if (seq selections)
+      (when-let [filtered (filter #(= (.getName %) n) selections)]
+        (-> filtered first)))))
+
+(defn full-selection [env fragments]
+  (let [selections  (->clj (get-selections-get env))
+        nodes (:nodes selections)
+        edges-node (some->> selections
+                            :edges
+                            (mapcat get-selections)
+                            (filter #(= (.getName %) "node"))
+                            first)
+        node-target-entity (traverse edges-node "target_entity")
+        node-source-entity (traverse edges-node "source_entity")]
+    (set (cond-> []
+           nodes (concat (mapcat #(resolve-fields % fragments) nodes))
+           edges-node (concat (resolve-fields edges-node fragments))
+           node-target-entity (concat (resolve-fields node-target-entity fragments))
+           node-source-entity (concat (resolve-fields node-source-entity fragments))))))
```

Comment:
In my tests it seems to do only half of the work, it will return only the immediate field selection and doesn't seem to bother resolving FragmentDefinitions from FragmentSpreads, I rewrote that part to make it recursive and start from it.